code/backend/batch/utilities/helpers/DocumentProcessorHelper.py

```diff
@@ -22,20 +28,38 @@ def __init__(self):
         pass
 
     def process(self, source_url: str, processors: List[Processor]):
-        vector_store_helper = AzureSearchHelper()
-        vector_store = vector_store_helper.get_vector_store()
-        for processor in processors:
+        env_helper: EnvHelper = EnvHelper()
+        if env_helper.AZURE_SEARCH_USE_INTEGRATED_VECTORIZATION:
             try:
-                document_loading = DocumentLoading()
-                document_chunking = DocumentChunking()
-                documents: List[SourceDocument] = []
-                documents = document_loading.load(source_url, processor.loading)
-                documents = document_chunking.chunk(documents, processor.chunking)
-                keys = list(map(lambda x: x.id, documents))
-                documents = [
-                    document.convert_to_langchain_document() for document in documents
-                ]
-                return vector_store.add_documents(documents=documents, keys=keys)
+                search_datasource_helper = AzureSearchIVDatasourceHelper(env_helper)
```

Comment:
I would suggest that we don't need to pass the env helper around now that it's a singleton, but happy to be overruled

Reply:
I think it's probably still worth it, it makes it easier to test, you can just pass the mock in rather than having the patch it in.